### PR TITLE
Homework 3 100%

### PR DESCRIPTION
**Pull request by gowgow1**
image

1. [2 point] - All functionality with back-end should be in services.js.  + 
2. [2 point] - Get courses from back-end. See API /courses/all in the SWAGGER. Save courses list to the 
store.  +
3. [2 point] - Get authors from back-end. See API /authors/all in the SWAGGER. Save authors list to the
store. +
4. [4 point] - Setup store:  + 
5. [1 point] - Get courses from store and render them into Courses component. +
6. [2 point] - After saving course in CreateCourse component course should be added to store. +
7. [2 point] - After deletion course in Courses component course should be deleted from store. +

### CODE REVIEW

File: courses-app/src/components/Courses/Courses.jsx [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
 
-const Courses = ({ togglePage, initCourses, initAuthors }) => {
+const Courses = () => {
 	const [InputValue, setInputValue] = useState('');
-	const [courses, setCourses] = useState(initCourses);
+	const coursesList = useSelector(getCourses);
+	const authorsList = useSelector(getAuthors);
+	const history = useHistory();
+	const redirectToAddCourse = () => {
+		history.push('/courses/add');
```
Comment (DrDiman) [inline courses-app/src/components/Courses/Courses.jsx:18]: nice to have: navigation file, where u can store each path for the navifation of the frontend + nice to have file with all api url for backend, where u are calling for data

File: courses-app/src/services/services.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+class Service {
+	_apiBase = 'http://localhost:3000';
+
+	default(body) {
```
Comment (DrDiman) [inline courses-app/src/services/services.js:4]: this function could be as:
```
async default(url, method, body) {
    let options = {
	method,
	headers: { 'Content-Type': 'application/json' }
    }
  
  if(body) {
    options = {
      ...options, 
      body: JSON.stringify(body)
    }
  }
  
  const response = await fetch(`${this._apiBase}${url}`, options)
  return response.json();
}
  ```

File: courses-app/src/store/authors/actionCreators.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import * as actions from './actionTypes';
+
+export const addAuthor = (item) => ({
+	type: actions.ADD_AUTHOR,
+	payload: { item },
```
Comment (DrDiman) [inline courses-app/src/store/authors/actionCreators.js:5]: no need to wrap into the separate object in every action

Comment (DrDiman) [reply courses-app/src/store/authors/actionCreators.js:5]: better as:

```
export const addAuthor = author => ({
	type: actions.ADD_AUTHOR,
	payload: author,
})
```

File: courses-app/src/store/courses/actionCreators.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import * as actions from './actionTypes';
+
+export const addCourse = (item) => ({
+	type: actions.ADD_COURSE,
+	payload: { item },
```
Comment (DrDiman) [inline courses-app/src/store/courses/actionCreators.js:5]: the same everywhere - as above

File: courses-app/src/store/user/reducer.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import * as actions from './actionTypes';
+
+const unLogin = { isAuth: false, name: '', email: '', token: '' };
```
Comment (DrDiman) [inline courses-app/src/store/user/reducer.js:3]: `isAuth` is redundant bool flag -> e.g. u can rely on just email or name

File: courses-app/src/store/courses/actionTypes.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export const ADD_COURSE = 'add_course';
```
Comment (DrDiman) [inline courses-app/src/store/courses/actionTypes.js:1]: use same convention as `export const ADD_COURSE = 'addCourse';`

File: courses-app/src/store/user/reducer.js (hunk 5 above)
Comment (gowgow1) [reply courses-app/src/store/user/reducer.js:3]: according by task's requirement

File: courses-app/src/services/services.js (hunk 2 above)
Comment (gowgow1) [reply courses-app/src/services/services.js:4]: done

**Pull request closed without merging**